security/firely-auth/firely-auth-sso.rst

```diff
@@ -4,24 +4,41 @@ Single Sign-On using external providers
 =======================================
 
 Firely Auth can use external OpenID Connect enabled systems for authenticating users instead of relying on a dedicated local user database.
-If one or more single sign-on (SSO) providers are configured, users are provided with the option in the login UI to use the external login provider instead of an additional username and password.
+If one or more single sign-on (SSO) providers are configured, users are provided with the option in the login UI to use the external login provider instead of an additional username and password. Note: if only a single SSO provider is configured, Firely Auth will not show a dedicated login screen but redirect to the external login automatically.
 
 Multiple configuration parts are necessary to enable SSO in Firely Auth:
 
 #. Configure Firely Auth as a client in the remote SSO system 
 
     Technically Firely Auth uses an implicit token workflow to access an ID token from the external system. 
     Make sure to generate an OAuth 2 client id and client secret for Firely Auth. Additionally, ensure that the client is allowed to access ID tokens.
+    Firely Auth must be allowed to allowed request an ``openid``, ``profile`` and ``email`` scope.
 
-#. Configure user accounts in the remote SSO system and in the local Firely Auth instance.. 
+#. Configure user accounts in the remote SSO provider
     
-    Each user account must be associated with a unique claim inside the ID token. 
-    After a successful SSO login, authenticated users are matched against password-less local user accounts to see if the accounts are provisioned to use the local Firely Auth instance. 
-    Admins must create an empty local user account assoicated with a matching claim. However it is not necessary to assign these accounts a password.
-    Automated provisioning of local user accounts based on a SSO login is not yet supported by Firely Auth.
+    Each user account must be associated with a unique claim inside the ID token.
+    For each SSO user Firely Auth keeps a local password-less account which acts as a statement that at some point in time the SSO has logged into Firely Auth.
+    This password-less account is auto-provisioned if the ``AllowAutoProvision`` setting is enabled for the SSO provider. Otherwise an admin needs to create the SSO account via the UI or REST API of Firely Auth.
 
 #. Configure all SSO details in the ``ExternalIdentityProviders`` configuration section
 
+    After a successful login accounts details are automatically updated based on the information provided by the SSO system. 
+    The local password-less account is matched on all logins against the SSO account. This matching is based on either an ``email`` claim or ``oid`` claim from the ID token. This way Firely Auth always contains an up-to-date user profile.
```

Comment:
Is it `oid` or `openid`? I see `openid` on the Microsoft Graph under API permissions.

Reply:
"oid" is the claim that you get back inside the token if you request "openid" as a scope. It's confusing, I admit.